builtin/credential/aws/path_role.go

```diff
@@ -41,14 +54,32 @@ that it must match the IAM role ARN specified by this parameter.
 The value is prefix-matched (as though it were a glob ending in
 '*').  The configured IAM user or EC2 instance role must be allowed
 to execute the 'iam:GetInstanceProfile' action if this is
-specified.`,
+specified. This is only checked when auth_type is
+ec2.`,
 			},
 			"bound_iam_instance_profile_arn": {
 				Type: framework.TypeString,
 				Description: `If set, defines a constraint on the EC2 instances to be associated
 with an IAM instance profile ARN which has a prefix that matches
 the value specified by this parameter. The value is prefix-matched
-(as though it were a glob ending in '*').`,
+(as though it were a glob ending in '*'). This is only checked when
+auth_type is ec2.`,
+			},
+			"inferred_entity_type": {
+				Type: framework.TypeString,
+				Description: `When auth_type is iam, the
+AWS entity type to infer from the authenticated principal. The only supported
+value is ec2_instance, which will extract the EC2 instance ID from the
+authenticated role and apply restrictions specific to EC2 instances (such as
+role_tag). The configured EC2 client must be able to find the inferred instance
```

Comment:
This documentation needs an update to remove the `role_tag` specifics. It might be a good idea to list out the bounds that gets checked when the inferred entity type is `ec2_instance`.